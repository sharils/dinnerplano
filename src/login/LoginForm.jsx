import PropTypes from "prop-types";
import React from "react";

export default function LoginForm({ email, password }) {
  return (
    <form>
      <label htmlFor="email">
        Email: <input type="email" id="email" {...email} />
      </label>
      <label htmlFor="password">
        Password: <input type="password" id="password" {...password} />
      </label>
      <button type="submit">Login</button>
    </form>
  );
}
LoginForm.propTypes = {
  email: PropTypes.shape({
    onChange: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired
  }).isRequired,
  password: PropTypes.shape({
    onChange: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired
  }).isRequired
};
