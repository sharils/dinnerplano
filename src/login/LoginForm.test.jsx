import React from "react";
import renderer from "react-test-renderer";
import LoginForm from "./LoginForm";

const getInputProps = ({ onChange = () => {}, value = "" } = {}) => ({
  onChange,
  value
});

it("renders correctly", () => {
  const tree = renderer
    .create(
      <LoginForm
        email={getInputProps({ value: "alice@example.com" })}
        password={getInputProps({ value: "alice's password" })}
      >
        Facebook
      </LoginForm>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
