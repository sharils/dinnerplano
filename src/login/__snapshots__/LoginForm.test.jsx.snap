// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <label
    htmlFor="email"
  >
    Email: 
    <input
      id="email"
      onChange={[Function]}
      type="email"
      value="alice@example.com"
    />
  </label>
  <label
    htmlFor="password"
  >
    Password: 
    <input
      id="password"
      onChange={[Function]}
      type="password"
      value="alice's password"
    />
  </label>
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
